internal/provider/data_source_gitlab_project_milestones.go

```diff
@@ -0,0 +1,129 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/mitchellh/hashstructure"
+	"github.com/xanzy/go-gitlab"
+)
+
+var _ = registerDataSource("gitlab_project_milestones", func() *schema.Resource {
+	validMilestoneStates := []string{"active", "closed"}
+
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_project_milestones`" + ` data source allows get details of a project milestones.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/milestones.html)`,
+
+		ReadContext: dataSourceGitlabProjectMilestonesRead,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
```

Comment:
Same reasoning as for the other schemas:

```suggestion
			"project": {
```

Reply:
Connect with previous discussion.